api/backend/blueflood/blueflood.go

```diff
@@ -132,20 +136,32 @@ func (b *blueflood) FetchSingleSeries(request api.FetchSeriesRequest) (api.Times
 	}
 	queryResolution := b.config.bluefloodResolution(
 		request.Timerange.Resolution(),
-		request.Timerange.Start())
+		request.Timerange.Start(),
+	)
 
+	// Sample the data at the given `queryResolution`
 	queryUrl, err := b.constructURL(request, sampler, queryResolution)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
-
-	// Issue GET to fetch metrics
 	parsedResult, err := b.fetch(request, queryUrl)
 	if err != nil {
 		return api.Timeseries{}, err
 	}
 
-	values := processResult(parsedResult, request.Timerange, sampler, queryResolution)
+	combinedResult := parsedResult.Values
+
+	// Sample the data at the FULL resolution.
+	// In order to do this, we use the same `request` object but hard-code the ResolutionFull parameter.
+	fullResolutionQueryURL, err := b.constructURL(request, sampler, ResolutionFull)
+	if err == nil {
```

Comment:
I'm not sure.

On the one hand, I don't expect this part to error. On the other hand, if it does, the query isn't really broken- if you're querying for low-resolution data it's a bit strange if the high-resolution data breaks it.

Also, from a practical effort standpoint, the mocking tests for Blueflood fail since they don't have a `constructURL` for the corresponding high-resolution fetch.